### PR TITLE
For pdf build

### PR DESCRIPTION
**Pull request by jfbu**
This PR is to make PDF build possible. I have tested it on v1.0.0.

- as Sphinx has now reached 1.7.2, and scipy's conf.py uses deprecated variables which were removed at 1.6, I have converted to 1.6 format (regarding LaTeX variables; but html-scipyorg build seems to be fine too; I will check in detail later if html variables are ok).

- to workaround issues with Unicode characters, the PDF will be built via XeLaTeX.

- unfortunately ≠ is unknown to XeLaTeX's default OpenType font, so extra set-up is added to handle it.

- I tried with DejaVu which does have the ≠, also it fits with the legends from included graphics. But the font is bigger than Latin Modern, and I had to increase line spacing to get better looking tables. This increases number of pages. Hence I reverted that pair of commits.

- the PR includes a backport of a fix to an autosummary bug which will be part of Sphinx 1.7.3.

- I fixed two math LaTeX mark-up bugs in the documentation sources, which were revealed from LaTeX "there is no \<control-char\> in font..." warnings in log.

See the commit messages for additional information.

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -167,50 +161,75 @@
...
 #latex_logo = None
 
-# For "manual" documents, if this is true, then toplevel headings are parts,
-# not chapters.
-#latex_use_parts = False
+# Documents to append as an appendix to all manuals.
+#latex_appendices = []
+
+# If false, no module index is generated.
+latex_use_modindex = False
```
Comment (pv) [inline doc/source/conf.py]: iiuc, this option was renamed to latex_domain_indices

Comment (jfbu) [reply doc/source/conf.py]: yes `latex_use_modindex` was removed at 1.6. Sorry for oversight I will fix later.

**Comment by pv** [comment]
Please re-file this PR against the master branch, not maintenance/1.0.x
I guess you need to close this PR, rebase, and re-file it.

**Comment by jfbu** [comment]
@pv Thanks, I think I can rebase on master, force push, and edit here to change the merge base.

Should I lower Sphinx requirement to 1.5 ? testing fails on CI because of 1.6 requirement.

**Comment by jfbu** [comment]
Regarding numpydoc v0.8.0, the references citations in PDF will work as hyperlinks but not be useful on printed paper (docstrings numbering is kept so there are about 300 duplicated  ``[1]`` which show up as keys in the PDF bibliography, which appears at end of it). Also, the docstrings do not have a copy in PDF on location of the actual references. But for use on screen, it is fine.

**Comment by pv** [comment]
Right, might be useful to open a numpydoc issue for the bad latex bibliography.

**Comment by jfbu** [comment]
I think I should close and reopen to hopefully trigger appveyor testing again?

Regarding PDF build with numpydoc 0.8.0, Latexmk will report about 600 duplicated references. This is only bad for printed out documentation. For using on screen in a PDF viewer, links to the bibliography work fine. There are no backlinks, but the PDF viewer should have a "go back" icon.

**Comment by pv** [comment]
(close+reopen to trigger travis)

**Comment by pv** [comment]
Thanks, seems to work OK here.

**Pull request merged**